src/machine/machine_stm32wle5.go

```diff
@@ -396,6 +398,27 @@ func (t *TIM) enableMainOutput() {
 	t.Device.BDTR.SetBits(stm32.TIM_BDTR_MOE)
 }
 
+//---------- TRNG related code
+
+func InitRNG() {
+	stm32.RCC.AHB3ENR.SetBits(stm32.RCC_AHB3ENR_RNGEN)
+	stm32.RNG.CR.SetBits(stm32.RNG_CR_RNGEN)
+}
+
+func ReadRNG() (uint32, error) {
+	cnt := RNG_MAX_READ_RETRIES
+	for cnt > 0 && !stm32.RNG.SR.HasBits(stm32.RNG_SR_DRDY) {
+		cnt--
+	}
+	if cnt == 0 {
+		return 0, errors.New("STM32WL TRNG timeout")
```

Comment:
Should probably declare this as a named `error` at some package level. Makes it possible to add godoc comments to it, and easier to test for when received from a call to `Read`. 

In the case of STM32, perhaps a _reusable_ declaration in `machine_stm32.go` or even `crypto/rand/rand_stm32.go` (changed to a generic "timeout" message, obviously)?

Reply:
That, and making it global avoids a possible heap allocation and probably reduces code size as well.